internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreName.java

```diff
@@ -0,0 +1,62 @@
+package com.linkedin.venice.meta;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+
+
+/**
+ * This class is a handle to refer to a store.
+ *
+ * It intentionally does not contain any operational state related to the referenced store.
+ *
+ * It is appropriate to use as a map key. Its {@link #equals(Object)} and {@link #hashCode()} delegate to the same
+ * functions on the String form of the store name.
+ *
+ * The purpose of using this handle class rather than a String is two-fold:
+ *
+ * - It is a stronger type than String, since a String can contain anything.
+ * - It can be more performant, since shared instances are allocated once and reused thus causing less garbage, and
+ *   shared instances are also faster to use as map keys (the hash code gets cached, and equality checks can be resolved
+ *   by identity, in the common case).
+ */
+public class StoreName {
+  private final String name;
+
+  /**
+   * Not intended to be called directly! Instead, use:
+   *
+   * {@link NameRepository#getStoreName(String)}
+   */
+  StoreName(String name) {
+    if (!Store.isValidStoreName(name)) {
```

Comment:
Should we move `isValidStoreName` from `Store` to `StoreName` class?

Reply:
Good idea, I've done that in my latest commit.